Assets/Scripts/Data/PickupPup/Currency/CurrencySystem.cs

```diff
@@ -93,11 +103,18 @@ public void ChangeCoins(int deltaCoins)
         ChangeCurrencyAmount(CurrencyType.Coins, deltaCoins);
     }
 
-    public void ChangeFood(int deltaFood)
+    public void ChangeFood(int deltaFood, bool isSpecial)
```

Comment:
Generally anywhere we're using a boolean for `isSpecial`, I would recommend replacing with an enum so we can more easily scale up to more types of dog food.